spring-integration-core/src/test/java/org/springframework/integration/dsl/transformers/TransformerTests.java

```diff
@@ -287,6 +342,71 @@ public IdempotentReceiverInterceptor idempotentReceiverInterceptor() {
 			return idempotentReceiverInterceptor;
 		}
 
+		@Bean
+		public PollableChannel subFlowTestReplyChannel() {
+			return new QueueChannel();
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow syncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle((p, h) -> someServiceGateway.deriveValueSync(p)))
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow asyncSubFlowEnricher(SomeServiceGateway someServiceGateway) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle((p, h) -> someServiceGateway.deriveValueAsync(p), ec -> ec.async(true)))
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		public MessageChannel enricherReplyChannel() {
+			return MessageChannels.direct().get();
+		}
+
+		@Bean
+		@Autowired
+		public IntegrationFlow terminatingSubFlowEnricher(SomeService someService) {
+			return f -> f
+					.enrich(e -> e.<TestPojo>requestPayload(p -> p.getPayload().getName())
+							.subFlow(sf -> sf
+									.<String>handle(m -> someService.aTerminatingServiceMethod(m)))
+							.replyChannel("enricherReplyChannel")
+							.<String>headerFunction("foo", m -> m.getPayload())
+							.propertyFunction("name", m -> m.getPayload()))
+					.channel("subFlowTestReplyChannel");
+		}
+
+		@Bean
+		public GatewayProxyFactoryBean someServiceGateway() {
+			GatewayProxyFactoryBean gateway = new GatewayProxyFactoryBean(SomeServiceGateway.class);
+			gateway.setAsyncExecutor(exec());
+			return gateway;
+		}
+
+		@Bean
+		public SomeService someService() {
+			return new SomeService();
+		}
+
+		@Bean
+		public AsyncTaskExecutor exec() {
+			SimpleAsyncTaskExecutor simpleAsyncTaskExecutor = new SimpleAsyncTaskExecutor();
```

Comment:
Let's remove this bean altogether!
The `GatewayProxyFactoryBean` uses its instance by default anyway:
```
private volatile AsyncTaskExecutor asyncExecutor = new SimpleAsyncTaskExecutor();
```
This test isn't about Messaging Gateway, so, would be better to make it as simple as possible.